src/main/java/org/testng/internal/TestListenerHelper.java

```diff
@@ -115,26 +117,31 @@ public static ListenerHolder findAllListeners(Class<?> cls, IAnnotationFinder fi
     result.listenerClasses = Lists.newArrayList();
 
     while (cls != Object.class) {
+      List<IListenersAnnotation> annotations = finder.findAnnotations(cls, IListenersAnnotation.class);
       IListenersAnnotation l = finder.findAnnotation(cls, IListenersAnnotation.class);
-      if (l != null) {
-        Class<? extends ITestNGListener>[] classes = l.getValue();
-        for (Class<? extends ITestNGListener> c : classes) {
-          result.listenerClasses.add(c);
-
-          if (ITestNGListenerFactory.class.isAssignableFrom(c)) {
-            if (result.listenerFactoryClass == null) {
-              result.listenerFactoryClass = (Class<? extends ITestNGListenerFactory>) c;
-            } else {
-              throw new TestNGException(
-                  "Found more than one class implementing "
-                      + "ITestNGListenerFactory:"
-                      + c
-                      + " and "
-                      + result.listenerFactoryClass);
+      annotations.add(l);
+      annotations.forEach(anno -> {
+        if (anno != null) {
```

Comment:
There is not supposed to have null values in the list

Reply:
If there has no annotation in super class and interfaces, it will cause NPE

Reply:
What do you think about:

```java
if (l != null) {
  annotations.add(l);
}
annotations.forEach(anno -> {
  ...
```

If no annotation in superclass, the list will be empty and no null will be added if no annotation in the class

Reply:
yes, I'll change it